### PR TITLE
Buttons & Links Refactor

### PR DESCRIPTION
**Pull request by APJohns**
Refactored the following component to reduce Inputs
- Action
- Landing Page Header

Added Dropdown component which is used in lph and is not yet exported from the package.

### CODE REVIEW

File: projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  [id]="uuid"
+  class="btn btn-primary btn-wide dropdown-toggle"
+  data-toggle="dropdown"
+  aria-haspopup="true"
+  aria-expanded="false"
+>
+  {{ dropdownText }}
+</button>
+<div
+  class="dropdown-menu dropdown-menu-right object-dropdown"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html:13]: We should rename 'object-dropdown' when we make this more generic.

File: projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  <bp-action
+    *ngFor="let dropwdownItem of dropdownItems"
+    classes="dropdown-item"
+    [text]="dropwdownItem.text"
+    [ariaLabel]="dropwdownItem.ariaLabel"
+    [href]="dropwdownItem.href"
+    [routerLink]="dropwdownItem.routerLink"
+    [target]="dropwdownItem.target"
+    [isDisabled]="dropwdownItem.isDisabled"
+    (click)="emitAction(dropwdownItem.action)"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html:25]: Typo in dropdown

File: projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  aria-expanded="false"
+>
+  {{ dropdownText }}
+</button>
+<div
+  class="dropdown-menu dropdown-menu-right object-dropdown"
+  [attr.aria-labelledby]="uuid"
+>
+  <bp-action
+    *ngFor="let dropwdownItem of dropdownItems"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html:17]: Typo in dropdown

File: projects/epsilon-blueprint/src/lib/landing-page-header/landing-page-header.component.html [hunk 4]
```diff
@@ -3,41 +3,22 @@
...
-      <a
-        *ngIf="actionElement === 'a'"
-        class="btn btn-primary btn-wide"
-        [attr.href]="href"
-      >
-        {{ actionText }}
-      </a>
+      <bp-action
+        *ngIf="!dropdownItems"
+        classes="btn btn-primary btn-wide"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/landing-page-header/landing-page-header.component.html:16]: We should switch btn-wide to btn-lg when we refactor

File: projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<button
+  dropdownToggle
+  type="button"
+  [id]="uuid"
+  class="btn btn-primary btn-wide dropdown-toggle"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/dropdown/dropdown.component.html:5]: We should switch btn-wide to btn-lg when we refactor

File: projects/epsilon-blueprint/src/lib/action/action.component.html [hunk 6]
```diff
@@ -1,30 +1,32 @@
...
->
-  <ng-container *ngTemplateOutlet="actionContent"></ng-container>
-</button>
+<ng-container [ngSwitch]="true">
+  <a
+    *ngSwitchCase="routerLink !== undefined"
+    [ngClass]="classes"
+    [attr.aria-label]="ariaLabel"
+    [routerLink]="routerLink"
+    [target]="target ? target: null"
```
[the remark this answers is not in the document]
Comment (APJohns) [reply projects/epsilon-blueprint/src/lib/action/action.component.html:7]: I've been suing it everywhere I notice the attribute remains `attribute=""` when nothing is passed to it.

**Pull request merged**